src/main/java/com/intel/gkl/compression/IntelDeflater.java

```diff
@@ -168,8 +176,13 @@ public void finish() {
      *         output buffer
      */
     @Override
-    public int deflate(byte[] b, int off, int len ) {
-
+    public int deflate(byte[] b, int off, int len ) throws NullPointerException {
+        if(b == null) {
+            throw new NullPointerException("Input is null");
+        }
+        if(len <= 0) {
+            throw new NullPointerException("Input buffer length is zero.");
```

Comment:
Can we check for null before value check? Also for value condition can use IllegalArgumentException to be consistent.

Reply:
Since Deflator is covered in PR 122, will undo the changes here.